src/shared/cloudformation/cloudformation.ts

```diff
@@ -749,3 +762,160 @@ export namespace CloudFormation {
+ */
+export async function refreshSchemas(extensionContext: vscode.ExtensionContext): Promise<void> {
+    CFN_SCHEMA_PATH = normalizeSeparator(path.join(extensionContext.globalStoragePath, 'cloudformation.schema.json'))
+    SAM_SCHEMA_PATH = normalizeSeparator(path.join(extensionContext.globalStoragePath, 'sam.schema.json'))
+    let manifest: string | undefined
+    try {
+        const manifestFetcher = new HttpResourceFetcher(MANIFEST_URL, { showUrl: true })
+        manifest = await manifestFetcher.get()
+    } catch (e) {
+        getLogger().error(`Failed getting manifest at ${MANIFEST_URL}:`, e)
+
+        return
+    }
+
+    if (!manifest) {
+        getLogger().error(`Schema manifest at ${MANIFEST_URL} was undefined`)
+
+        return
+    }
+
+    try {
+        const details = getManifestDetails(manifest)
+
+        await getRemoteOrCachedFile({
+            filepath: CFN_SCHEMA_PATH,
+            version: details.version,
+            url: details.cfnUrl,
+            cacheKey: 'cfnSchemaVersion',
+        })
+        await getRemoteOrCachedFile({
+            filepath: SAM_SCHEMA_PATH,
+            version: details.version,
+            url: details.samUrl,
+            cacheKey: 'samSchemaVersion',
+        })
+    } catch (e) {
+        getLogger().error('Could not get details from manifest:', (e as Error).message)
+    }
+}
+
+/**
+ * Maps a template file to a CFN or SAM schema.
+ * If present, removes association with the other type of schema.
+ * Does not modify other schemas not managed by AWS.
+ * @param path Template file path
+ * @param type Template type to use for filepath
+ */
+export async function updateYamlSchemasArray(
+    path: string,
+    type: 'cfn' | 'sam',
+    config: WorkspaceConfiguration = vscode.workspace.getConfiguration('yaml'),
+    paths: { cfnSchema: string; samSchema: string } = { cfnSchema: CFN_SCHEMA_PATH, samSchema: SAM_SCHEMA_PATH }
+): Promise<void> {
+    const relPath = normalizeSeparator(getWorkspaceRelativePath(path) ?? path)
+    const schemas: { [key: string]: string | string[] | undefined } | undefined = config.get('schemas')
+    const writeTo = type === 'cfn' ? paths.cfnSchema : paths.samSchema
+    const deleteFrom = type === 'sam' ? paths.cfnSchema : paths.samSchema
+    let newWriteArr: string[] = [relPath]
+    let newDeleteArr: string[] = []
+
+    if (schemas) {
+        if (schemas[writeTo]) {
+            newWriteArr = Array.isArray(schemas[writeTo])
+                ? (schemas[writeTo] as string[])
+                : [schemas[writeTo] as string]
+            if (!newWriteArr.includes(relPath)) {
+                newWriteArr.push(relPath)
+            }
+        }
+        if (schemas[deleteFrom]) {
+            const temp = Array.isArray(schemas[deleteFrom])
+                ? (schemas[deleteFrom] as string[])
+                : [schemas[deleteFrom] as string]
+            newDeleteArr = temp.filter(val => val !== relPath)
+        }
+    }
+
+    try {
```

Comment:
What happens if the user deletes a template? To me it looks like there would be a dangling association to a non-existent file since this function is only called when a template is loaded. 

If we're writing to the user's settings, we should at least keep it clean.

Reply:
I'll check. The file watcher's glob pattern might pick up the change on delete while the IDE is open but it definitely wouldn't do it if the file was deleted with the IDE closed. Could be worth doing a full audit on IDE load as well.